lib/apiservers/engine/backends/eventmonitor.go

```diff
@@ -218,34 +218,32 @@ func (p DockerEventPublisher) PublishEvent(event plevents.BaseEvent) {
 
 	vc := cache.ContainerCache().GetContainer(event.Ref)
 	if vc == nil {
-		log.Warnf("Portlayer event for container %s but not found in cache", event.Ref)
+		log.Warnf("Received portlayer event %s for container %s but not found in cache", event.Event, event.Ref)
 		return
 	}
 
 	var attrs map[string]string
-	// TODO: move to a container.OnEvent() so that container drives the necessary changes
-	// based on event activity
+	// TODO: move to a container.OnEvent() so that container drives the necessary changes based on event activity
 	switch event.Event {
 	case plevents.ContainerStopped,
-		plevents.ContainerPoweredOff:
-		// since we are going to make a call to the portLayer lets execute this in a
-		// go routine
+		plevents.ContainerPoweredOff,
+		plevents.ContainerFailed:
+		// since we are going to make a call to the portLayer lets execute this in a go routine
 		go func() {
 			attrs := make(map[string]string)
 			// get the containerEngine
 			code, err := NewContainerBackend().containerProxy.exitCode(vc)
 			if err != nil {
 				// log the error, but continue
-				log.Errorf("unable to get exitCode for die event: %s", err)
+				log.Warnf("Unable to get exit code for %s: %s", vc.ContainerID, err)
```

Comment:
This may be expected in the container failed case - if so can we avoid logging it as it will confuse matters in the logs.